### PR TITLE
Preparing for 0.12.0.rc0: Final changes before RC

### PR DESCRIPTION
**Pull request by mbaijal**
## Description ##
Preparing for 0.12.0.rc0: Final changes before RC

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] For user-facing API changes, API doc string has been updated.
- [x] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Updated the NEWS.md to match release notes with updates from new PRs
- [x] Added the 0.12.0 release tag to the NEWS.md. 

## Comments ##
The 0.12.0 tag does not exist at the moment but I believe this is the right way to do it since code must not be edited after RC. 
There was a comment about not using the 0.11.0.rc* tag during last release here - https://lists.apache.org/thread.html/171c0ff73e5f7539611a8024fef42968f5ffea800c354b2cdfa04ab2@%3Cgeneral.incubator.apache.org%3E
This is also documented here - https://github.com/apache/incubator-mxnet/issues/7750

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,34 +1,46 @@
...
+  - Improved ImageRecordIO image loading performance and added indexed RecordIO support.
+  - Added better openmp thread management to improve CPU performance.
+### New Features - Gluon
+  - Added enhancements to the Gluon package, a high-level interface designed to be easy to use while keeping most of the flexibility of low level API. Gluon supports both imperative and symbolic programming, making it easy to train complex models imperatively with minimal impact on performance. Neural networks (and other machine learning models) can be defined and trained with `gluon.nn` and `gluon.rnn` packages. 
+  - Added new loss functions - `SigmoidBinaryCrossEntropyLoss`, `CTCLoss`, `HuberLoss`, `HingeLoss`, `SquaredHingeLoss`, `LogisticLoss`, `TripletLoss`.
   - `gluon.Trainer` now allows reading and setting learning rate with `trainer.learning_rate` property.
-  - Added `mx.autograd.grad` and experimental second order gradient support (though most operators don't support second order gradient yet)
-  - Added `ConvLSTM` etc to `gluon.contrib`
+  - Added API `HybridBlock.export` for exporting gluon models to MXNet format.
+  - Added `ConvLSTM` to gluon.contrib.
```
Comment (szha) [inline NEWS.md]: Convolutional recurrent networks for RNN, LSTM, and GRU. VariationalDropout.

File: NEWS.md [hunk 2]
```diff
@@ -1,34 +1,46 @@
...
+  - Added `mx.autograd.grad` and experimental second order gradient support (most operators don't support second order gradient yet).
   - Autograd now supports cross-device graphs. Use `x.copyto(mx.gpu(i))` and `x.copyto(mx.cpu())` to do computation on multiple devices.
+### New Features - Sparse Tensor Support
+  - Added support for sparse matrices. 
+  - Added limited cpu support for two sparse formats in `Symbol` and `NDArray` - `CSRNDArray` and `RowSparseNDArray`.
+  - Added a sparse dot product operator and many element-wise sparse operators.
+  - Added a data iterator for sparse data input - `LibSVMIter`.
+  - Added three optimizers for sparse gradient updates: `Ftrl`, `SGD` and `Adam`.
+  - Added `push` and `row_sparse_pull` with `RowSparseNDArray` in distributed kvstore.
+  - Better support for IDE auto-completion. IDEs like PyCharm can now correctly parse mxnet operators.
```
Comment (szha) [inline NEWS.md]: This is not a sparse feature.

File: NEWS.md (hunk 1 above)
Comment (sebouh) [reply NEWS.md]: @szha can you clarify what needs to be added here?

File: NEWS.md (hunk 2 above)
Comment (mbaijal) [reply NEWS.md]: moved to other features

File: NEWS.md (hunk 1 above)
Comment (szha) [reply NEWS.md]: Please update line 14 with the features that I listed. It's not just conv lstm.

Comment (mbaijal) [reply NEWS.md]: @szha Can you please review this - https://cwiki.apache.org/confluence/display/MXNET/MXNet+0.12.0+Release+Notes

I am copying changes from here into NEWS.md. Once release notes are reviewed I will update the PR. I do not want to trigger multiple PR builds right now

Comment (szha) [reply NEWS.md]: OK. Commented.

Comment (sebouh) [reply NEWS.md]: what about VariationalDropout?

**Pull request merged**